mpxe/sims/mci.py

```diff
@@ -0,0 +1,47 @@
+from mpxe.protogen import stores_pb2
+from mpxe.protogen import mcp2515_pb2
+from mpxe.protogen import gpio_pb2
+
+import time
+
+from mpxe.sims import sim
+
+GPIO_KEY = (stores_pb2.MxStoreType.GPIO, 0)
+MCP2515_KEY = (stores_pb2.MxStoreType.MCP2515, 0)
+
+class Mci(sim.Sim):
+    def __init__(self):
+        super(Mci, self).__init__()
+        self.gpio_ab = None
+        self.tx_id = 0
+        self.tx_data = 0
+    def handle_update(self, pm, proj):
+        def store_gpio_ab():
+            self.gpio_ab = [proj.stores[GPIO_KEY].state[i] for i in range(32)]
+        if self.gpio_ab == None:
+            store_gpio_ab()
+        elif proj.stores[GPIO_KEY].state[9] == 1 and self.gpio_ab[9] != 1:
+            time.sleep(0.1)
+            store_gpio_ab()
+            # update precharge monitor and latch out
+            self.set_gpio(proj, 'b', 0, True)
+            time.sleep(0.1)
+            self.set_gpio(proj, 'a', 10, True)
+        else:
+            store_gpio_ab()
```

Comment:
The gpio interfacing stuff here is kinda ugly, there should probably be a few more layers of abstraction; I don't really have any idea what this is doing without diving into the mci schematic